pkg/trace/sampler/sampler.go

```diff
@@ -53,6 +53,15 @@ const (
 // SamplingPriority is the type encoding a priority sampling decision.
 type SamplingPriority int8
 
+// samplingMechanism is the source of the sampling rates.
+// Possible values
+//     1: agent rate (Default)
+//     2: dynamically calculated remote rate
+//     6: remote rate defined by user
+//     7: remote rate defined by Datadog
+// This list is not exhaustive.
+type samplingMechanism uint32
```

Comment:
Can we create constants for these?
```
const (
    samplingMechanismAgent samplingMechanism = 1
    samplingMechanismDynamic samplingMechanism = 2
    samplingMechanismRemoteUser samplingMechanism = 6
    samplingMechanismRemote samplingMechanism = 7
)
```
...with each being documented and explained.

Also, please search the entire repository and use these everywhere. Also, use the `samplingMechanism` type everywhere where `uint32` is used today please.

Reply:
As discussed offline, we decided not to add `const`s. I used `SamplingMechanism` everywhere.